flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -471,8 +476,9 @@ private static ListStateDescriptor<SortedMap<Long, byte[]>> buildStateDescriptor
     return new ListStateDescriptor<>("iceberg-files-committer-state", sortedMapTypeInfo);
   }
 
-  static long getMaxCommittedCheckpointId(Table table, String flinkJobId, String operatorId) {
-    Snapshot snapshot = table.currentSnapshot();
+  static long getMaxCommittedCheckpointId(
+      Table table, String flinkJobId, String operatorId, String branch) {
+    Snapshot snapshot = table.snapshot(branch);
```

Comment:
the fetched snapshot doesn't seem being used.

Reply:
It does after line 484. This is where we start from the head of the branch (snapshot in this case) and traverse the lineage until we find a snapshot produced by the same flink job and get the most recently committed checkpoint

Reply:
sorry, I misread the diff. thought it was a new line/variable